src/lib/proxy/index.ts

```diff
@@ -55,14 +56,19 @@ export const incomingRequestHandler = async (req, res, next) => {
     // i.e. replace ${auth.accessToken} from the integration template
     // with the authentication access token retrieved from the database.
     const forwardedHeaders = headersToForward(req.rawHeaders)
-    const { url, headers } = await buildRequest({
+    let { url, headers } = await buildRequest({
       authentication,
       integration,
       method: req.method,
       forwardedHeaders: forwardedHeaders,
       path: req.originalUrl.substring(('/proxy/' + integrationName).length + 1)
     })
 
+    //If there is a header in the request specifying the MailChimp datacenter, replace the "server" in the request URL with the datacenter name.
+    if (chimpDC) {
```

Comment:
Out of curiosity, is it a pattern that you've seen if other APIs? If so, we could build a more universal support for this in the future!

Reply:
So far I have only seen this in the MailChimp integration. I think this is a MailChimp-specific problem because they use different data centers for different users. I will raise it up if I see anyone else behaving like this.

Reply:
Understood, thanks for the details!